spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/BasicAcknowledgeablePubsubMessage.java

```diff
@@ -36,9 +43,15 @@
 	PubsubMessage getPubsubMessage();
 
 	/**
-	 * Accessor for the subscription source of the Pub/Sub message.
-	 * @return the name of the subscription
+	 * Acknowledge (ack) the message asynchronously
+	 * @return ListenableFuture&lt;String&gt;
 	 */
-	String getSubscriptionName();
+	ListenableFuture<String> ack();
+
+	/**
+	 * Negatatively achnowledge (nack) the message asynchronously
+	 * @return ListenableFuture&lt;String&gt;
```

Comment:
We need some description what the `ListenableFuture` is about. Also consider to use a `{@code }` instead of this ugly manual encoding